src/devices_models/devices/common/set_expression.jl

```diff
@@ -0,0 +1,17 @@
+"""
+Replaces an expression value in the expression container if the key exists
+"""
+function set_expression!(
+    container::OptimizationContainer,
+    ::Type{S},
+    cost_expression::JuMP.AbstractJuMPScalar,
+    component::T,
+    time_period::Int,
+) where {S <: CostExpressions, T <: PSY.Component}
+    if has_container_key(container, S, T)
```

Comment:
The behavior in the case where the key doesn't exist could be surprising to a user. Is that really the best thing to do? Should there be a log message at least?